src/library/scala/collection/immutable/ImmutableArray.scala

```diff
@@ -45,7 +45,7 @@ sealed abstract class ImmutableArray[+A]
     val dest = Array.ofDim[Any](length)
     Array.copy(unsafeArray, 0, dest, 0, length)
     dest(index) = elem
-    ImmutableArray.unsafeWrapArray(dest)
+    ImmutableArray.unsafeWrapArray(dest.asInstanceOf[Array[B]])
```

Comment:
Can we make the new casts in this file do the correct cast of the resulting `ImmutableArray` to set a proper example?